pkg/appgw/backendaddresspools.go

```diff
@@ -45,9 +47,43 @@ func (c appGwConfigBuilder) getPools(cbCtx *ConfigBuilderContext) []n.Applicatio
 		allPools = append(allPools, *addr)
 	}
 
+	if cbCtx.EnvVariables.EnableBrownfieldDeployment == "true" {
```

Comment:
This is the feature flag- *off* by default so this code is not executed unless we add `APPGW_ENABLE_BROWNFIELD_DEPLOYMENT="true"` env variable